File: RosService/src/package/src/service_server.cpp
#include "ros/ros.h"
#include "package/count_magnitude.h"
#include <cmath>

bool add(package::count_magnitude::Request &req,package::count_magnitude::Response &resp) {
    // mengolah data
    resp.magnitude = sqrt(req.komponenx*req.komponenx + req.komponeny*req.komponeny + req.komponenz*req.komponenz);


    // mencetak hasil data
    ROS_INFO("Nilai komponen X : %ld" , (long int) req.komponenx);
    ROS_INFO("Nilai komponen Y : %ld" , (long int) req.komponeny);
    ROS_INFO("Nilai komponen Z : %ld" , (long int) req.komponenz);
    ROS_INFO("Nilai magnitude : %lf\n", (float) resp.magnitude);
    return true;
}

int main(int argc, char **argv){
    // inisialisasi node
    ros::init(argc,argv,"service_server");

    // inisialisasi handler node
    ros::NodeHandle nh;

    // inisialisasi server dari handler
    ros::ServiceServer server = nh.advertiseService("count_magnitude",add);
    ROS_INFO("Ready to count the magnitude.");
    ros::spin();

    return 0;
}

File: RosService/src/package/src/service_server.py
#!/usr/bin/env python

import rospy
from package.srv import count_magnitude

def count(req):
    # Process and print data
    magnitude = (req.komponenx**2 + req.komponeny**2 + req.komponenz**2)**0.5
    rospy.loginfo("Nilai komponen X : %d" % req.komponenx)
    rospy.loginfo("Nilai komponen Y : %d" % req.komponeny)
    rospy.loginfo("Nilai komponen Z : %d\n" % req.komponenz)
    return magnitude

def count_magnitude_server():
    # Initialize node
    rospy.init_node("server")

    # Initialize server
    s = rospy.Service('count_magnitude', count_magnitude, count)
    rospy.loginfo("Ready to count the magnitude!")
    rospy.spin()

if __name__ == '__main__':
    count_magnitude_server()
